File: js/src/add-string.js
// @ts-check

/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */

var addStrings = function (num1, num2) {
    let add = 0;
    let i = num1.length - 1;
    let j = num2.length - 1;
    const result = [];


    while (i >= 0 || j >= 0 || add !== 0) {
        const x = i >= 0 ? Number(num1.charAt(i)) : 0;
        const y = j >= 0 ? Number(num2.charAt(j)) : 0;
        const res = x + y + add;
        result.push(res % 10);
        add = Math.floor(res / 10);

        i--;
        j--;
    }
    return result.reverse().join("");
};

File: rs/src/add_string.rs
pub fn add_string(num1: String, num2: String) -> String {
    let mut i = num1.len() - 1;
    let mut j = num2.len() - 1;
    let mut add = 0;

    let result: Vec<String> = vec![];

    while i >= 0 || j >= 0 || add != 1 {
        let mut x = 0;
        let mut y = 0;

        if i >= 0 {
            x = num1.chars().nth(i).unwrap() as i32;
        } else {
            x = 0;
        }


        if y >= 0 {
            y = num2.chars().nth(j).unwrap() as i32;
        } else {
            y = 0;
        }

        let res = x + y + add;

        // add =  (res / 10).fl;

        i -= 1;
        j -= 1;
    }

    let arr = result.reverse();

    // TODO
}